src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Threading;
+
+using Microsoft.Win32.SafeHandles;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public sealed class ECDsaOpenSsl : ECDsa
+    {
+        public ECDsaOpenSsl()
+            : this(521)
+        {
+        }
+
+        public ECDsaOpenSsl(int keySize)
+        {
+            KeySize = keySize;
+            _key = new Lazy<SafeEcKeyHandle>(GenerateKey);
+        }
+
+        /// <summary>
+        /// Create an ECDsaOpenSsl from an existing <see cref="IntPtr"/> whose value is an
+        /// existing OpenSSL <c>EC_KEY*</c>.
+        /// </summary>
+        /// <remarks>
+        /// This method will increase the reference count of the <c>EC_KEY*</c>, the caller should
+        /// continue to manage the lifetime of their reference.
+        /// </remarks>
+        /// <param name="handle">A pointer to an OpenSSL <c>EC_KEY*</c></param>
+        /// <exception cref="ArgumentException"><paramref name="handle" /> is invalid</exception>
+        public ECDsaOpenSsl(IntPtr handle)
+        {
+            if (handle == IntPtr.Zero)
+                throw new ArgumentException(SR.Cryptography_OpenInvalidHandle, "handle");
+
+            SafeEcKeyHandle ecKeyHandle = SafeEcKeyHandle.DuplicateHandle(handle);
+
+            // Set base.KeySize to avoid throwing an extra Lazy at the GC when
```

Comment:
I don't see what this comment is pertaining to; there aren't any calls to base.KeySize here...